### PR TITLE
json alerts part 2

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR continues to work towards json alerts.

### CODE REVIEW

File: skyportal/utils/gcn.py [hunk 1]
```diff
@@ -266,46 +280,61 @@ def get_skymap_url(root, notice_type, timeout=10):
...
-        retraction = int(retraction.attrib['value'])
-        if retraction == 1:
-            return True
+    if isinstance(root, dict):
+        retraction = root.get('retraction')
+    else:
+        retraction = root.find("./What/Param[@name='Retraction']")
+        if retraction is not None:
+            retraction = int(retraction.attrib['value'])
+            if retraction == 1:
```
Comment (Theodlz) [inline skyportal/utils/gcn.py:289]: Idk if it's git but it looks like in the JSON case you don't return True based on anything (the if statement looking at the retraction value is in the else that's above). Is that on purpose?

File: skyportal/utils/gcn.py [hunk 2]
```diff
@@ -266,46 +280,61 @@ def get_skymap_url(root, notice_type, timeout=10):
...
+        if loc is None:
+            return ra, dec, error
+
+        ra = loc.find('./AstroCoords/Position2D/Value2/C1')
+        dec = loc.find('./AstroCoords/Position2D/Value2/C2')
+        error = loc.find('./AstroCoords/Position2D/Error2Radius')
 
-    if None in (ra, dec, error):
-        return ra, dec, error
+        if None in (ra, dec, error):
```
Comment (Theodlz) [inline skyportal/utils/gcn.py:313]: same comment for this method

File: skyportal/utils/notifications.py [hunk 3]
```diff
@@ -65,24 +69,27 @@ def gcn_notification_content(target, session):
...
+            ra = notice_content.get("ra")
+            dec = notice_content.get("dec")
+            error = notice_content.get("ra_dec_error")
+        else:
+            loc = notice_content.find('./WhereWhen/ObsDataLocation/ObservationLocation')
+            ra = loc.find('./AstroCoords/Position2D/Value2/C1')
+            dec = loc.find('./AstroCoords/Position2D/Value2/C2')
+            error = loc.find('./AstroCoords/Position2D/Error2Radius')
+
+            if ra is not None:
```
Comment (Theodlz) [inline skyportal/utils/notifications.py:82]: same comment here. Any reason not to apply this dtype conversion to both cases?

File: alembic/versions/b854bf5d22b3_json_alerts.py [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column('gcnevents', sa.Column('gcn_type', sa.String(), nullable=True))
+    op.alter_column(
+        'gcnnotices',
+        'notice_type',
+        existing_type=postgresql.ENUM(
```
Comment (Theodlz) [inline alembic/versions/b854bf5d22b3_json_alerts.py:26]: not familiar with this `existing_type` arg here. What is this supposed to do?

File: skyportal/handlers/api/gcn.py [hunk 5]
```diff
@@ -171,7 +172,11 @@ def post_gcnevent_from_xml(
 
     if event is None:
         event = GcnEvent(
-            dateobs=dateobs, sent_by_id=user_id, trigger_id=trigger_id, aliases=aliases
+            dateobs=dateobs,
+            sent_by_id=user_id,
+            trigger_id=trigger_id,
+            aliases=aliases,
+            gcn_type="voevent",
```
Comment (Theodlz) [inline skyportal/handlers/api/gcn.py:179]: Quick question. Why do we need to track this on the event? I mean to me JSON/VOEVENT is purely for the notice and shouldn't matter for this event itself.

Comment (Theodlz) [reply skyportal/handlers/api/gcn.py:179]: We could have both voevent notices and JSON notices for the same event

**Review by Theodlz — COMMENTED** [review]
Just a few questions

File: services/gcn_service/gcn_service.py [hunk 6]
```diff
@@ -171,9 +172,6 @@ def poll_events(*args, **kwargs):
                                 notify=False,
                             )
                         elif alert_type == "json":
-                            # save the payload to a file
-                            with open(f'/tmp/{str(uuid.uuid4())}.json', 'w') as f:
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply services/gcn_service/gcn_service.py:175]: No worries.

File: skyportal/utils/gcn.py (hunk 1 above)
Comment (mcoughlin) [reply skyportal/utils/gcn.py:291]: Fixed, I think.

File: skyportal/utils/gcn.py (hunk 2 above)
Comment (mcoughlin) [reply skyportal/utils/gcn.py:315]: The function returns ra, dec, error at the end.

File: skyportal/utils/notifications.py (hunk 3 above)
Comment (mcoughlin) [reply skyportal/utils/notifications.py:82]: same deal here.

File: skyportal/handlers/api/gcn.py (hunk 5 above)
Comment (mcoughlin) [reply skyportal/handlers/api/gcn.py]: Ok I moved it over to the notice.

**Comment by Theodlz** [comment]
Thanks for addressing the comments @mcoughlin. WIll give another review soon

**Pull request merged**